react/src/components/ResourceAllocationFormItems.tsx

```diff
@@ -177,38 +181,35 @@ const ResourceAllocationFormItems: React.FC<
         const acceleratorResourceOfPreset = _.omitBy(
           preset.resource_slots,
           (value, key) => {
-            if (['mem', 'cpu', 'shmem'].includes(key) || value === '0')
-              return true;
+            if (['mem', 'cpu', 'shmem'].includes(key)) return true;
           },
         );
         if (currentImageAcceleratorLimits.length === 0) {
+          // When current image doesn't require any accelerator,
+          // It's available if the preset doesn't have any accelerator
           if (_.isEmpty(acceleratorResourceOfPreset)) {
             return true;
           } else {
             return false;
           }
-        }
-
-        return _.some(currentImageAcceleratorLimits, (limit) => {
-          return _.some(acceleratorResourceOfPreset, (value, key) => {
+        } else {
+          // When current image requires some accelerator,
+          // It's available if the preset has a required accelerator value that is larger than the current image's minimum value
+          return _.some(currentImageAcceleratorLimits, (limit) => {
             return (
-              limit?.key === key && _.toNumber(value) >= _.toNumber(limit?.min)
+              limit?.key &&
+              acceleratorResourceOfPreset[limit?.key] &&
+              _.toNumber(acceleratorResourceOfPreset[limit?.key]) >=
+                _.toNumber(limit?.min)
             );
           });
-        });
+        }
       },
     ).map((preset) => preset.name);
-
-    return currentImageAcceleratorLimits.length > 0
+    return currentImageAcceleratorLimits.length === 0
```

Comment:
If `currentImageAcceleratorLimits` is not empty, `allocatablePresetNames` should be determined using `byImageAcceleratorLimits`.